src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -138,38 +137,39 @@ public Payment getPlatformPayment(String name) {
      * @return None if the platform does not exist
      */
     public CompletableFuture<Boolean> publishTask(String name, Experiment experiment) throws TaskOperationException {
-
-        TaskRecord record = new TaskRecord();
-        record.setExperiment(experiment.getId());
-        record.setStatus(TaskStatus.running);
-        record.setCrowdPlatform(name);
-
-        TaskRecord result = tasksOps.createTask(record);
-        if (result == null)
-            throw new TaskOperationException("Task could not be created");
+        ExperimentsPlatformRecord record = experimentsPlatformOps.getExperimentsPlatform(name, experiment.getId()).
+                orElseThrow(() -> new IllegalStateException("Platform is not activated for experiment " + experiment));
+
+        BiFunction<String, Throwable, Boolean> handlePublishResult = (s1, throwable) -> {
```

Comment:
it makes the method easier to read, because we can avoid the nesting. Just style, no other big reason